int32 target_index
---
int32 target_value
---
int32 current_index
int32 current_value
